### PR TITLE
OCLOMRS-35: As a user I should be able to search and see all sources

### PR DESCRIPTION
**Pull request by emasys**
# JIRA TICKET NAME:
[As a user I should be able to search and see all sources](https://issues.openmrs.org/browse/OCLOMRS-35)

# Summary:
Implement the UI and logic for retrieving all sources in OCL, search for a specific source, filter by source types and sort by ascending or descending order of names.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,3 +26,4 @@ yarn.lock
 
 #test snapshots
 **/__snapshots__/**
+src/components/dashboard/container/ListContainer.jsx
```
[the remark this answers is not in the document]
Comment (emasys) [reply .gitignore]: Yeah, fixed.

[the remark this answers is not in the document]
Comment (shakirandagire) [reply .gitignore]: LGTM

File: src/tests/Dashboard/action/index.test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    const expectedActions = [
+      { type: IS_FETCHING, payload: true },
+      { type: FETCH_SOURCES, payload: [sources] },
+      { type: IS_FETCHING, payload: false },
+    ];
+
+    const store = mockStore({ payload: {} });
+
+    return store.dispatch(fetchSources('ciel', ['Dictionary'], 10, 1, 'sortAsc=name')).then(() => {
+      // return of async actions
```
Comment (dkayiwa) [inline src/tests/Dashboard/action/index.test.js]: Do we need the above comment?

**Comment by emasys** [comment]
Fixed @dkayiwa

File: src/tests/Dashboard/action/index.test.js (hunk 2 above)
Comment (emasys) [reply src/tests/Dashboard/action/index.test.js]: Not really. 
It has been deleted.

**Pull request merged**